WooCommerce/Classes/ViewRelated/Orders/Order Creation/EditableOrderViewModel.swift

```diff
@@ -1323,6 +1318,20 @@ extension EditableOrderViewModel {
         })
         stores.dispatch(action)
     }
+
+    /// Validates if the given productID can be found in local storage, and updates the Order with the correspondent product or variation.
+    /// Display a notice error otherwise.
+    ///
+    private func updateOrderWithProduct(_ productID: Int64) {
+        if let product = allProducts.first(where: { $0.productID == productID }) {
+            orderSynchronizer.setProduct.send(.init(product: .product(product), quantity: 1))
+        } else if let productVariation = allProductVariations.first(where: { $0.productVariationID == productID }) {
```

Comment:
I find it a bit confusing that we are naming the function `updateOrderWithProduct` and the parameter `productID` but we also handle variations and variation ids, could we perhaps name it in a more generic way to handle product variations as well?

Reply:
As we changed the signature to `updateOrderWithProduct(_ product: Product)` in https://github.com/woocommerce/woocommerce-ios/pull/9768/commits/c38d3a4d644d3cb82da3c0f398b74a205cda1389 , I left the name as it is, but happy to change it to something else if you think should be clearer 👍